app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = User.includes(posts: :comments).find(params[:user_id])
     @posts = @user.posts.order(created_at: :desc)
   end
 
   def show
-    @user = User.find(params[:user_id])
-    @post = Post.find(params[:id])
+    @post = Post.includes(:author, :comments).find(params[:id])

```

Comment:
- [ ] Please while testing for the views and fixing N + 1 problems, You have some of the tests failing so make sure all your tests all passing to ensure that the application functions perfectly. You can see the failing test below :point_down: 
  
  ![Screenshot from 2023-04-13 01-12-39](https://user-images.githubusercontent.com/92303891/231623036-a6e65c49-8026-40fc-93ca-841489944173.png)